napari/_qt/layer_controls/qt_labels_controls.py

```diff
@@ -23,7 +24,16 @@
 from ..widgets.qt_mode_buttons import QtModePushButton, QtModeRadioButton
 from .qt_layer_controls_base import QtLayerControls
 
-INT32_MAX = 2 ** 31 - 1
+
+def double_as_uint_spinbox(dtype: np.dtype):
```

Comment:
rather than a factory function, maybe we should just make a special subclass?

```python
class LongLongSpinBox(QDoubleSpinBox):
    def __init__(self, dtype, parent=None):
        # important to be able to pass a parent to the widget
        super().__init__(parent)
        iinfo = np.iinfo(dtype)
        self._dtype = dtype
        self.setMinimum(max(iinfo.min, 0))
        ...
```

Reply:
also if the intention of this is to "behave" as an integer spinbox that just has a bigger range, we should make sure that it always returns integers:

```python
    ....
    def value(self):
        return int(super().value())
```